File: 20-Valid-Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let array = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] == ')' && array.length > 0 && array[array.length - 1] === '('){
            array.pop();
            continue;
        }
        if (s[i] == '}' && array.length > 0 && array[array.length - 1] === '{'){
            array.pop();
            continue;
        }
        if (s[i] == ']' && array.length > 0 && array[array.length - 1] === '['){
            array.pop();
            continue;
        }
        array.push(s[i]);
    }
    return array.length === 0 ? true : false;
};

File: 20-Valid-Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in range(len(s)):
            if s[i] == '(' or s[i] == '{' or s[i] == '[':
                stack.append(s[i])
                continue
            if not stack:
                return False

            if s[i] == ')' and stack[-1] == '(':
                stack.pop()
            elif s[i] == '}' and stack[-1] == '{':
                stack.pop()
            elif s[i] == ']' and stack[-1] == '[':
                stack.pop()
            else:
                return False
        return not stack